### PR TITLE
Feature/add filter analysis framework to show recently used framework

### PR DESCRIPTION
**Pull request by RobertKoikai**
Addresses : Add a filter in analysisFramework query to show most recently used frameworks #1350 
Depends on  the-deep/server

## Changes
Update the AnalysisFrameworkFilterSet to show Recently Used Framework

## This PR doesn't introduce any:

- [x]  temporary files, auto-generated files or secret keys
- [x]  n+1 queries
- [x]  flake8 issues
- [x]  `print`
- [x]  typos
- [x]  unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

**Review by thenav56 — CHANGES_REQUESTED** [review]
Looks good.

Let's also push the new graphql schema
https://github.com/the-deep/server/actions/runs/6250258951/job/16968710026?pr=1369
https://github.com/the-deep/server/blob/develop/.github/workflows/ci.yml#L49-L58
Fri Sep 22 11:47:38 AM +0545 2023

```bash
docker-compose exec web ./manage.py graphql_schema --out schema.graphql
```

Now it should be fine to add tests.

File: apps/analysis_framework/tests/test_filters.py [hunk 1]
```diff
@@ -21,3 +26,29 @@ def test_search_filter(self):
...
+            modified_at=timezone.now() - timedelta(days=30),
+            lead=lead2
+        )
+
+        obtained = self.filter_class(data={'recently_used': True}).qs
+        expected = [af1, af2]
+
+        obtained_list = sorted(list(obtained), key=lambda x: x.title)
+        expected_list = sorted(list(expected), key=lambda x: x.title)
+        self.assertEqual(obtained_list, expected_list)
```
Comment (thenav56) [inline apps/analysis_framework/tests/test_filters.py]: Let's add some more data here.
```suggestion
        af1, af2, af3, _ = AnalysisFrameworkFactory.create_batch(3)

        lead = LeadFactory.create_batch(2)
        now = timezone.now()
        # Within date range window
        EntryFactory.create(
            analysis_framework=af1,
            lead=lead,
            modified_at=now - timedelta(days=90),
        )
        EntryFactory.create(
            analysis_framework=af2,
            lead=lead,
            modified_at=now - timedelta(days=30),
        )
        # Outside date range window
        EntryFactory.create(
            analysis_framework=af3,
            lead=lead,
            modified_at=now - timedelta(days=190),
        )
        # Make sure we only get af1, af2
        obtained = set(list(
            self.filter_class(data={'recently_used': True}).qs.values_list('id', flat=True)
        ))
        expected = set([af1.pk, af2.pk])
        self.assertEqual(obtained, expected)
```

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/analysis_framework/tests/test_filters.py (hunk 1 above)
Comment (RobertKoikai) [reply apps/analysis_framework/tests/test_filters.py]: @thenav56 ,
Did you run these changes on your end ?
I get `ValueError: not enough values to unpack (expected 4, got 3)` & `ValueError: Cannot assign "[<Lead: Lead-0>, <Lead: Lead-1>]": "Entry.lead" must be a "Lead" instance`

Comment (thenav56) [reply apps/analysis_framework/tests/test_filters.py]: @RobertKoikai No, it is just a change request with a pseudo suggestion.
Use this instead for the error you mentioned
```python3
        af1, af2, af3, _ = AnalysisFrameworkFactory.create_batch(4)
        lead = LeadFactory.create()
```

File: apps/lead/tests/test_filters.py
Comment (thenav56) [inline apps/lead/tests/test_filters.py:1]: Hey @RobertKoikai Any reason on why you are changing this file?

**Review by thenav56 — CHANGES_REQUESTED** [review]

**Review by thenav56 — APPROVED** [review]

**Pull request merged**